src/agent/src/rpc.rs

```diff
@@ -1722,6 +1747,150 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
     }
 }
 
+fn pull_image_from_registry(image: &str, src_creds: &str) -> Result<()> {
+    let source_image = format!("{}{}", "docker://", image);
+
+    // TODO LATER derive directory from image registry/namespace/repository/tag to make this more dynamic
+    // Define the target tranport and path for the manifest image, with signature
+    let target_path_manifest = format!("{}{}", "dir://", IMAGE_MANIFEST);
+
+    // Define the target transport and path for the OCI image, without signature
+    let target_path_oci = format!("{}{}", "oci://", IMAGE_OCI);
+
+    // Create directory into which to copy the manifest image
+    let status = Command::new("mkdir")
+        .arg("-v")
+        .arg("-p")
+        .arg(&IMAGE_MANIFEST)
+        .status()
+        .expect("Cannot create directory");
+
+    assert!(status.success());
+
+    // Create directory into which to copy the OCI image
+    // Will add code to munge the image registry/namespace/repository/tag to make this more dynamic
+    let status = Command::new("mkdir")
+        .arg("-v")
+        .arg("-p")
+        .arg(&IMAGE_OCI)
+        .status()
+        .expect("Cannot create directory");
+
+    assert!(status.success());
+
+    // TODO - fix up this nasty hack - skopeo won't accept blank credentials, so set them to something
+    let creds = if src_creds.trim().is_empty() {
+        "<blank>"
+    } else {
+        src_creds
+    };
+
+    let status = Command::new(SKOPEO_PATH)
+        .arg("copy")
+        .arg(source_image)
+        .arg(&target_path_manifest)
+        .arg("--src-creds")
+        .arg(creds)
+        .status()
+        .expect("Failed to pull image");
+
+    assert!(status.success());
+
+    // Copy image from one local file-system to another
+    // Resulting image is still stored in manifest format, but no longer includes the signature
+    // The image with a signature can then be unpacked into a bundle
+    let status = Command::new(SKOPEO_PATH)
+        .arg("copy")
+        .arg(&target_path_manifest)
+        .arg(&target_path_oci)
+        .arg("--remove-signatures")
+        .status()
+        .expect("Failed to copy image");
+
+    assert!(status.success());
```

Comment:
I'd rather see us properly propagating the `status()` call `Result` through, with a map_err and anyhow! for example. The error from `status` is an `ExitStatus` that implements `Display` fwiw.